topgg/lib/api.lua

```diff
@@ -1,199 +1,235 @@
 
-      if data.errors then
-         msg = parseErrors({msg}, data.errors);
-      end
-   end
-end
-   return nil, msg;
-end
-
-function Api:postStats(stats)
-   if not stats or (not stats.serverCount and not stats.server_count) then
-      error('Server count missing');
-   end
+  if method ~= 'GET' then
+    body = body and json.encode(body) or '{}'
 
-   if type(stats.serverCount) ~= 'number' and type(stats.server_count) ~= 'number' then
-      error("'serverCount' must be a number");
-   end
+    table.insert(request, { 'Content-Type', 'application/json' })
+    table.insert(request, { 'Content-Length', #body })
+  end
 
-   local __stats = {
-      server_count = stats.serverCount or stats.server_count,
-   };
+  local data, err = self:__commit(method, url, request, body)
 
-   if (stats.shardId or stats.shard_id) and (stats.shardCount or stats.shard_count) then
-     __stats.shard_id = stats.shard_id or stats.shardId
-     __stats.shard_count = stats.shard_count or stats.shardCount
-   end
-
-   local _, res = self:request('POST', '/bots/stats', __stats);
-   return res;
+  if data then
+    return data
+  else
+    return nil, err
+  end
 end
 
-function Api:getStats(id)
-   if type(id) ~= 'string' then
-      error("argument 'id' must be a string");
-   end
+function Api:__commit(method, url, request, body)
+  local success, res, msg = pcall(http.request, method, url, request, body)
+
+  if not success then
+    return nil, res
+  end
+
+  for i, v in ipairs(res) do
+    res[v[1]:lower()] = v[2]
+    res[i] = nil
+  end
+
+  local data =
+    res['content-type']:find('json', 1, true) and json.decode(
+      msg,
+      1,
+      json.null
+    ) or msg
+
+  if res.code < 300 then
+    return data, nil
+  elseif type(data) == 'table' then
+    if (data.code or data.status) and (data.message or data.detail) then
+      msg = string.format('HTTP Error %i : %s', data.code or data.status, data.message or data.detail)
+    else
+      msg = 'HTTP Error'
+    end
+
+    if data.errors then
+      msg = parse_errors({ msg }, data.errors)
+    end
+  end
+
+  return nil, msg
+end
 
-   local stats = self:request('GET', f('/bots/%s/stats', id));
+function Api:post_bot_server_count(server_count)
+  if type(server_count) ~= 'number' or server_count <= 0 then
+    error("'server_count' must be a number and non-zero")
+  end
 
-   return stats;
+  return self:__request('POST', '/bots/stats', { server_count = server_count })
 end
 
-function Api:getBot(id)
-   if type(id) ~= 'string' then
-      error("argument 'id' must be a string");
-   end
+function Api:get_bot_server_count()
+  local stats = self:__request('GET', '/bots/stats')
 
-   return self:request('GET', f('/bots/%s', id));
+  return stats and stats.server_count
 end
 
-function Api:getBots(query)
-   if query then
-      if type(query.fields) == 'table' then
-         query.fields = concat(query.fields, ',');
-      end
+function Api:get_bot(id)
```

Comment:
avoid renaming the public API please

Reply:
[Done](https://github.com/null8626/lua-sdk/commit/799f02b7819553740c9bd1da2011087b73317d04), with the exception of `Api:init(token);` to `local client = Api:new(token);`.